### PR TITLE
feat: vmmap blocknumber hashlookup

### PR DESCRIPTION
**Pull request by canonbrother**
## Summary

- re-enable vmmap block number dvm <-> evm
- auto infer is not included in this PR

## RPCs

- vmmap

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [x] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [ ] Includes consensus refactors
  - [x] None

### CODE REVIEW

**Comment by canonbrother** [comment]
BLOCKING: will do test after evm tests are fixed

File: lib/ain-rs-exports/src/evm.rs [hunk 1]
```diff
@@ -592,6 +592,27 @@ pub fn evm_try_get_block_number_by_hash(result: &mut ffi::CrossBoundaryResult, h
...
 }
 
+pub fn evm_try_get_dvm_block_number_by_number(
+    result: &mut ffi::CrossBoundaryResult,
+    number: u64,
+) -> u64 {
+    match SERVICES.evm.storage.get_block_by_number(&U256::from(number)) {
+        Ok(Some(block)) => {
+            let extra_data = block.header.extra_data;
+            let data = String::from_utf8(extra_data.to_vec()).unwrap();
```
Comment (canonbrother) [inline lib/ain-rs-exports/src/evm.rs:602]: the only dvm info for evm 
ref: https://github.com/DeFiCh/ain/blob/9bfd6bb665340674f793614341fed1d1ee276b35/lib/ain-evm/src/evm.rs#L341

File: lib/ain-rs-exports/src/evm.rs [hunk 2]
```diff
@@ -592,6 +592,27 @@ pub fn evm_try_get_block_number_by_hash(result: &mut ffi::CrossBoundaryResult, h
...
+            let Some(data) = data.split_whitespace().nth(1) else {
+                return cross_boundary_error_return(result, format!("Expected `DFI: <n>`, actual: {:?}", data));
```
Comment (prasannavl) [inline lib/ain-rs-exports/src/evm.rs:604]: This is miner set for the outside world as hint only. Not for internal use.

**Comment by canonbrother** [comment]
refer to #2361

**Pull request closed without merging**